api/v1alpha1/common_types.go

```diff
@@ -13,36 +13,48 @@ type ObjectMeta struct {
 	// (scope and select) objects. May match selectors of replication controllers
 	// and services.
 	// More info: http://kubernetes.io/docs/user-guide/labels
-	// +optional
+	// +kubebuilder:validation:Optional
 	Labels map[string]string `json:"labels,omitempty"`
 
 	// Annotations is an unstructured key value map stored with a resource that may be
 	// set by external tools to store and retrieve arbitrary metadata. They are not
 	// queryable and should be preserved when modifying objects.
 	// More info: http://kubernetes.io/docs/user-guide/annotations
-	// +optional
+	// +kubebuilder:validation:Optional
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
 
 type ControlPlaneEndpointSpec struct {
 	// The hostname on which the API server is serving.
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
```

Comment:
```suggestion
	// The hostname on which the API server is serving.
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:Required
```

Reply:
Why do you want it this way round? Alphabetically ordered? I'd prefer to have Required/Optional first and validation after. I can go through and make that consistent if you agree.

Reply:
I made required/optional consistently first.